subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/repositories/AuthenticationSupporter.java

```diff
@@ -145,6 +148,8 @@ private void populateAuthenticationCredentials() {
             return Cast.uncheckedCast(PasswordCredentials.class);
         } else if (AwsCredentials.class.isAssignableFrom(implType)) {
             return Cast.uncheckedCast(AwsCredentials.class);
+        } else if (HttpHeaderCredentials.class.isAssignableFrom(implType)) {
+            return Cast.uncheckedCast(HttpHeaderCredentials.class);
         } else {
             throw new IllegalArgumentException(String.format("Unknown credentials implementation type: '%s' (supported types: %s and %s).", implType.getName(), DefaultPasswordCredentials.class.getName(), DefaultAwsCredentials.class.getName()));
```

Comment:
`DefaultHttpHeaderCredentials` is missing from the exception message.

Reply:
Thanks, fixed.